### DISCUSSION TITLE
Errors only showing up after second submit

### DISCUSSION THREAD

**Discussion by hui444** [react-hook-form/react-hook-form#5304] [Q&A]
I am currently using version 6. The `errors` shows up when i console.log them after the input code, but when i console.log them in the submit method, it will show up empty until the submit button is pressed a second time. May I know if there is something I am missing? Or if there is a way around this?
Here is a [sandbox](https://codesandbox.io/s/react-hook-form-errors-validatecriteriamode-all-forked-p3xft?file=/src/index.js) that shows the issue

Any help would be appreciated!

More information: 
I am trying to make a 3 part expandable form, so if there is an error in the first part, on submit I want to expand the first part. However, since I am unable to detect the error on first submit in the submit method, I am unable to do so. :(

**Comment by bluebill1049** [comment]
`formState: { errors}` that's the missing part. have a look at this page: https://react-hook-form.com/migrate-v6-to-v7

**Reply by hui444** [reply]
Hi! I tried that and now the errors dont show at all
```
export default function App() {
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = useForm();

  const onSubmit = (e) => {
    e.preventDefault();
    console.log(errors);
    handleSubmit((data) => {
      console.log(data);
    })();
  }; ...
```

**Reply by bluebill1049** [reply]
<img width="1023" alt="Screen Shot 2021-05-24 at 5 49 06 pm" src="https://user-images.githubusercontent.com/10513364/119314669-57531880-bcb8-11eb-8990-f432bc43f69b.png">

if you are using v6, then you can read errors from useForm.

**Reply by bluebill1049** [reply]
if you want errors to show up with onChange, change `mode` to onChange, or onBlur. that will change the errors triggering behaviour.

**Reply by hui444** [reply]
May I know how I can change the `mode`? And my issue is the error not showing up in the onSubmit method, since I need the `errors` in there to trigger other actions.
